example/spec/parallel/ingestGranule/IngestUMMGSuccessSpec.js

```diff
@@ -0,0 +1,273 @@
+      providerModel.delete(provider),
+      executionModel.delete({ arn: workflowExecution.executionArn }),
+      granulesApiTestUtils.deleteGranule({
+        prefix: config.stackName,
+        granuleId: inputPayload.granules[0].granuleId
+      })
+    ]);
+  });
+
+  xit('completes execution with success status', () => {
+    expect(workflowExecution.status).toEqual('SUCCEEDED');
+  });
+
+  // This is a sanity check to make sure we actually generated UMM and also
+  // grab the location of the UMM file to use when testing move
+  describe('the processing task creates a UMM file', () => {
+    let processingTaskOutput;
+    let ummFiles;
+
+    beforeAll(async () => {
+      processingTaskOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'FakeProcessing');
+
+      ummFiles = processingTaskOutput.payload.filter((file) => file.includes('.cmr.json'));
+      s3ummJsonFileLocation = ummFiles[0];
+    });
+
+    it('creates a UMM JSON file', () => {
+      expect(ummFiles.length).toEqual(1);
+
+      const xmlFiles = processingTaskOutput.payload.filter((file) => file.includes('.cmr.xml'));
+      expect(xmlFiles.length).toEqual(0);
+    });
+  });
+
+  xdescribe('the PostToCmr task', () => {
+    let cmrResource;
+    let cmrLink;
+    let response;
+    let files;
+    let resourceHrefs;
+
+    beforeAll(async () => {
+      postToCmrOutput = await lambdaStep.getStepOutput(workflowExecution.executionArn, 'PostToCmr');
+      if (postToCmrOutput === null) throw new Error(`Failed to get the PostToCmr step's output for ${workflowExecution.executionArn}`);
+
+      files = postToCmrOutput.payload.granules[0].files;
+      cmrLink = postToCmrOutput.payload.granules[0].cmrLink;
+      cmrResource = await getOnlineResources(cmrLink);
+      response = await got(cmrResource[1].href);
+
+      resourceHrefs = cmrResource.map((resource) => resource.href);
+    });
+
+    it('has expected payload', () => {
+      granule = postToCmrOutput.payload.granules[0];
+      expect(granule.published).toBe(true);
+      expect(granule.cmrLink.startsWith('https://cmr.uat.earthdata.nasa.gov/search/granules.json?concept_id=')).toBe(true);
+
+      // Set the expected cmrLink to the actual cmrLink, since it's going to
+      // be different every time this is run.
+      const updatedExpectedpayload = cloneDeep(expectedPayload);
+      updatedExpectedpayload.granules[0].cmrLink = postToCmrOutput.payload.granules[0].cmrLink;
+
+      expect(postToCmrOutput.payload).toEqual(updatedExpectedpayload);
+    });
+
+    it('publishes the granule metadata to CMR', () => {
+      const result = conceptExists(granule.cmrLink);
+
+      expect(granule.published).toEqual(true);
+      expect(result).not.toEqual(false);
+    });
+
+    it('updates the CMR metadata online resources with the final metadata location', () => {
+      const distEndpoint = config.DISTRIBUTION_ENDPOINT;
+      const extension1 = urljoin(files[0].bucket, files[0].filepath);
```

Comment:
are we guaranteed there's only one file in the granules payload?  And that it's public?  should we filter for the actual file we want to check?

Reply:
I copied this out of the other ingest tests. We are guaranteed in this case, because we set up the test.